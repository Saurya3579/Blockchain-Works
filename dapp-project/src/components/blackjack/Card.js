import React, { Component } from 'react';

class Card extends Component {
    render() {
        var bgUrl = (this.props.hidden)
            ? 'url(../../images/hidden.png)'
            : 'url(../../images/' + this.props.face + '.png)';
        /* in react we pass the css properties as an object with camelCase variables referring to the respective CSS variables */
        var cardStyle = { backgroundImage: bgUrl };

        return (
            <div className='card' style={cardStyle} />
        );
    }
}

export default Card;
