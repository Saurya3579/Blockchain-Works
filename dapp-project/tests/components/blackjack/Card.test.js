import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Card from '../../../src/components/blackjack/Card';

test('should render NoMatch component', () => {
    const wrapper = shallow(<Card />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});