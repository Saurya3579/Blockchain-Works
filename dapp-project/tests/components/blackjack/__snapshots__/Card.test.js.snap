exports[`test should render NoMatch component 1`] = `
<div>
  <img
    alt=""
    src="./img/undefined.png" />
</div>
`;
